micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java

```diff
@@ -33,7 +33,13 @@ public MicrometerMetrics(MeterRegistry registry) {
             .publishPercentileHistogram()
             .register(registry);
     try {
-      final var result = timer.record(execution::execute);
+      final var result = timer.record(() -> {
+        try {
+          return execution.execute();
+        } catch (Exception e) {
+          throw new IllegalStateException(e);
```

Comment:
Why throw an `IllegalStateException` here?

Reply:
Changed it to `OperatorException`.